src/models/sssd_conf.py

```diff
@@ -1,34 +1,94 @@
-from .file import File
-from pathlib import Path
 from configparser import ConfigParser
+from pathlib import Path
+import os
+from shutil import copy2
 from SCAutolib import logger
+from SCAutolib.src import TEMPLATES_DIR, LIB_BACKUP
+from .file import File
 
 
 class SSSDConf(File):
-    _template = Path()
+    _template = Path(TEMPLATES_DIR, "sssd.conf")
     _sssd_conf_path = Path("/etc/sssd/sssd.conf")
-    _parser = ConfigParser()
-    # Needed to avoid parsing problems on inserting some 'specific' value,
-    # that cause an exception with default method
-    _parser.optionxform = str
-    with _sssd_conf_path.open() as f:
-        _parser.read_file(f)
+    _backup_original = Path(LIB_BACKUP, 'original_sssd.conf')
+    _backup_default = Path(LIB_BACKUP, 'default_sssd.conf')
+
+    @classmethod
+    def create(cls):
+        """
+        Creates sssd.conf file if not present in the system or update existing
+        sssd.conf file with values from template file provided by this package
+        and backup original file.
+        """
+        parser = ConfigParser()
+        # avoid problems with inserting some 'specific' values
+        parser.optionxform = str
+
+        # this method should be executed only once to create default config
+        # existing backup files indicates that this method was already executed
+        # in such case exception should be raised
```

Comment:
Not sure if an exception should be raised in all cases. I would suggest adding an option to not raising an exception something like `force: bool = False` as an argument that would mean: if `True`, no exception would be raised if a backup exists, just silently re-create sssd.conf, using existing backup and the template.

This is a topic that should be discussed, but I think this step in current implementation can cause some unexpected and hidden bugs/errors.